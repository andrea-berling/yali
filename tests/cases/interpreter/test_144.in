class Wizard {
  castSpell(spell) {
    // Methods should be able to accept a parameter
    print "Casting a magical spell: " + spell;
  }
}
class Dragon {
  // Methods should be able to accept multiple<|SPACE|> // parameters
  breatheFire(fire, intensity) {
    print "Breathing " + fire + " with intensity: "
      + intensity;
  }
}
var merlin = Wizard();
var smaug = Dragon();
if (false) {
  var action = merlin.castSpell;
  action("Fireball");
} else {
  var action = smaug.breatheFire;
  action("Fire", "100");
}

