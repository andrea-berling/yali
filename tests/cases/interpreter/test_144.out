Breathing Fire with intensity: 100
